Pokedex/Pokedex/ViewControllers/PokedexTableVC.m

```diff
@@ -0,0 +1,61 @@
+//
+//  PokedexTableVC.m
+//  Pokedex
+//
+//  Created by Cora Jacobson on 12/12/20.
+//
+
+#import "PokedexTableVC.h"
+#import "Pokedex-Swift.h"
+#import "Pokemon.h"
+#import "DetailVC.h"
+
+@interface PokedexTableVC ()
+
+@property (nonatomic) NSArray<Pokemon *> *pokemons;
+
+@end
+
+@implementation PokedexTableVC
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    [PokemonAPI.sharedController fetchAllPokemonWithCompletion:^(NSArray<Pokemon *> *pokemonArray, NSError *error) {
+        if (error) {
+            NSLog(@"Error fetching pokemon list: %@", error);
+        }
+        self.pokemons = pokemonArray;
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [self.tableView reloadData];
+        });
+    }];
+}
+
+#pragma mark - Table view data source
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return self.pokemons.count;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"pokemonCell" forIndexPath:indexPath];
+    cell.textLabel.text = _pokemons[indexPath.row].name.capitalizedString;
+    return cell;
+}
+
+#pragma mark - Navigation
+
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+    if ([segue.identifier isEqualToString:@"showDetailSegue"]) {
+        DetailVC *detailVC = (DetailVC *)segue.destinationViewController;
+        Pokemon *pokemon = [_pokemons objectAtIndex:[self.tableView indexPathForSelectedRow].row];
+        detailVC.pokemon = pokemon;
+        if (pokemon.identifier == nil) {
+            [self willChangeValueForKey:@"spriteURL"];
+            [PokemonAPI.sharedController fillInDetailsFor:pokemon];
+            [self didChangeValueForKey:@"spriteURL"];
```

Comment:
So I don't believe you need to call willChangeValueForKey and didChangeValueForKey, my understanding is any key value coding compliant class will get automatic notifications, so as long as you follow the naming convention that Lambda teaches you should not need to call these normally. I think where you would need to call these is for a more custom implementation of notifications where maybe you want to receive multiple notifications at once. I don't believe the network call to retrieve the Pokemon info is even finished when didChangeValueForKey runs.